app/routes.py

```diff
@@ -1 +1,255 @@
+def delete_one_goal(goal_id):
+
+    goal = validate_object("goal", goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response(jsonify({"details" : f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"}))
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+
+    request_body = request.get_json()
+    try:
+        new_task = Task.create(request_body)
+    except KeyError:
+        return make_response(jsonify({"details" : "Invalid data"}), 400)
+
+    if request_body.get("completed_at"):
+        new_task.is_complete = True
+        new_task.completed_at = datetime.now()
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = { "task": new_task.to_json()}
+
+    return make_response(response, 201)
+
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    title_query = request.args.get("sort")
+
+    if title_query:
+        tasks = Task.query.order_by(text(f'title {title_query}'))
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(
+            task.to_json()
+        )
+    return make_response(jsonify(tasks_response), 200)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_object("task", task_id)
+
+    response = { "task": task.to_json()}
+
+    return make_response(response, 200)
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+
+    task = validate_object("task", task_id)
+
+    request_body = request.get_json()
+
+    task.update(request_body)
+
+    db.session.commit()
+
+    response = { "task": task.to_json()}
+    return make_response(response, 200)
+
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+
+    task = validate_object("task", task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response(jsonify({"details" : f"Task {task.task_id} \"{task.title}\" successfully deleted"}))
+
+
+@tasks_bp.route("<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_completed(task_id):
+    task = validate_object("task", task_id)
+
+    task.is_complete = True
+    task.completed_at = datetime.now()
+
+    db.session.commit()
+
+    URL = "https://slack.com/api/chat.postMessage"
+    PARAMS = {'text' : f"Someone just completed the task {task.title}",
+    'channel' : 'task-notifications'}
+    HEADERS = {'Authorization' : f'Bearer {os.environ.get("TASKLIST_BOT_KEY")}'}
+    r = requests.get(url = URL, params = PARAMS, headers = HEADERS)
+
+    response = { "task": task.to_json()}
+
+    return response, 200
+
+    
+@tasks_bp.route("<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incompleted(task_id):
+    task = validate_object("task", task_id)
+
+    task.is_complete = False
+    task.completed_at = None
+
+    db.session.commit()
+
+    response = { "task": task.to_json()}
+
+    return response, 200
```

Comment:
keep consistency with your return statements here, so that users can use your API in a predictable fashion

```suggestion
    return make_response(jsonify(response), 200)
```